pkg/skaffold/schema/versions_test.go

```diff
@@ -482,10 +482,13 @@ func format(t *testutil.T, configs []string, apiVersions []string) string {
 
 func withLocalBuild(ops ...func(*latest.BuildConfig)) func(*latest.SkaffoldConfig) {
 	return func(cfg *latest.SkaffoldConfig) {
-		b := latest.BuildConfig{BuildType: latest.BuildType{LocalBuild: &latest.LocalBuild{Concurrency: &constants.DefaultLocalConcurrency}}}
+		b := latest.BuildConfig{BuildType: latest.BuildType{LocalBuild: &latest.LocalBuild{}}}
 		for _, op := range ops {
 			op(&b)
 		}
+		if len(b.Artifacts) > 0 {
+			b.LocalBuild.Concurrency = &constants.DefaultLocalConcurrency
+		}
```

Comment:
What if we instead pushed this to the local builder itself, so that it determined the default concurrency at build time?

(And isn't this duplicating what is being done in the call to `withLocalBuild()` in defaults.go?)

Reply:
this is a function in the test file that sets up the _expected_ config

Reply:
thought about pushing this to `local.Builder` but it doesn't maintain the `[]*latest.Artifact` slice and it seemed more convoluted to first set the default to 1 but later ignore it in the method.

Reply:
Couldn't we just change this method in the local builder to return 1?
https://github.com/GoogleContainerTools/skaffold/blob/89b83154e58d7982d7db510a70f273971206c27a/pkg/skaffold/build/local/local.go#L60-L65

Reply:
If we don't set it in `default.Set` then we'll need a reference to the `[]*latest.Artifact` per config in `local.Builder` to decide if it should return 0 or 1 by default (0 if len(artifacts) == 0, otherwise 1). That plumbing seems more convoluted that this fix.

Reply:
Oh I didn't clue in that this definition was in a test — I thought it was the implementation of `withLocalBuilder` in pkg/skaffold/schema/defaults/ 🤦